test/linalg.jl

```diff
@@ -0,0 +1,64 @@
+using NiSparseArrays:imul!
+using SparseArrays, Random, Test
+Random.seed!(1234)
```

Comment:
Set the seed inside each `@testset` rather than here to make the tests having consistent results (reproducible).

Reply:
FYI, perhaps surprisingly, I vaguely remember that `@testset` sets its own random see during initialization...

```julia
julia> Random.seed!(1234)
TaskLocalRNG()

julia> @testset begin
           @show rand()
       end;
rand() = 0.7601155742719967

julia> @testset begin
           @show rand()
       end;
rand() = 0.7601155742719967
```

Reply:
Thanks for suggestions from mentors(I haven't considered the impact of random seed in/outside testset on the results before). So maybe there is no need to set the seed in each testset?

Reply:
There're gripes on this testset-owned random seed behavior, so I would go as @GiggleLiu suggested. But generally, if your algorithm is robust enough, you don't need to care about the random seed value.